stdlib/public/core/Collection.swift

```diff
@@ -1199,18 +1199,18 @@ extension Collection {
       return []
     }
 
-    var result = ContiguousArray<T>()
-    result.reserveCapacity(n)
-
-    var i = self.startIndex
-
-    for _ in 0..<n {
-      result.append(try transform(self[i]))
-      formIndex(after: &i)
+    return try unsafe Array<T>(unsafeUninitializedCapacity: n) { (
+      storage: inout UnsafeMutableBufferPointer<T>,
+      initializedCount: inout Int
+    ) throws(E) -> Void in
+      var collectionIndex = self.startIndex
+      for bufferIndex in 0 ..< n {
```

Comment:
Let's avoid capturing state that's also available from our parameters:
```suggestion
      for bufferIndex in storage.indices {
```

Reply:
Oh good point, also cleaner.